sounddevice.py

```diff
@@ -2657,20 +2675,26 @@ def _split(value):
     return invalue, outvalue
 
 
-def _check(err, msg=""):
+def _check(err, msg=''):
     """Raise error for non-zero error codes."""
-    if err < 0:
-        msg += ': ' if msg else ''
-        if err == _lib.paUnanticipatedHostError:
-            info = _lib.Pa_GetLastHostErrorInfo()
-            hostapi = _lib.Pa_HostApiTypeIdToHostApiIndex(info.hostApiType)
-            msg += 'Unanticipated host API {0} error {1}: {2!r}'.format(
-                hostapi, info.errorCode, _ffi.string(info.errorText).decode())
-        else:
-            msg += _ffi.string(_lib.Pa_GetErrorText(err)).decode()
-        raise PortAudioError(msg)
-    return err
+    if err >= 0:
+        return err
+
+    if msg: msg += ': '
+
+    if err == _lib.paUnanticipatedHostError:
+        info = _lib.Pa_GetLastHostErrorInfo()
+        hostapi = _lib.Pa_HostApiTypeIdToHostApiIndex(info.hostApiType)
+        apiinfo = _lib.Pa_GetHostApiInfo(hostapi)
+        apiname = _ffi.string(apiinfo.name).decode() if apiinfo else 'unknown'
+        hosterrmsg = _ffi.string(info.errorText).decode()
+        msg += "Unanticipated {0} host error {1}".format(apiname, info.errorCode)
+        if hosterrmsg:
+            msg += ": '{0}'".format(hosterrmsg)
+    else:
+        msg += _ffi.string(_lib.Pa_GetErrorText(err)).decode()
```

Comment:
I guess both are possible, but is it better to obtain the error details here or move it to the `PortAudioError` constructor?

I'm kind of leaning to the latter ...